glue/sales_val_flagging.py

```diff
@@ -430,109 +438,43 @@ def sql_type_to_pd_type(sql_type):
-
-    # Process standard deviation
-    df_std = (
-        df.assign(
-            rolling_window=lambda x: pd.to_datetime(
-                x["rolling_window"], format="%Y%m"
-            ).dt.date
-        )
-        .assign(group=lambda x: x[stat_groups].astype(str).apply("_".join, axis=1))
-        .assign(
-            group_mean=lambda x: x.groupby("group")["meta_sale_price"].transform("mean")
-        )
-        .assign(
-            std=lambda x: (x["meta_sale_price"] - x["group_mean"]) / x[std_dev_col_name]
-        )
-    )
-
-    df_std_cols = ["group", "std"]
-    if not condos:
-        df_std = df_std.assign(
-            group_mean_sqft=lambda x: x.groupby("group")["sv_price_per_sqft"].transform(
-                "mean"
-            ),
-            std_sqft=lambda x: (x["sv_price_per_sqft"] - x["group_mean_sqft"])
-            / x[std_dev_sqft_col_name],
-        )
-        df_std_cols.append("std_sqft")
-
-    df_std = df_std[df_std_cols].drop_duplicates(subset="group", keep="first")
-
-    # Process group sizes and means
+    # Calculate group sizes
     group_sizes = df.groupby(stat_groups).size().reset_index(name="group_size")
     df = df.merge(group_sizes, on=stat_groups, how="left")
 
-    unique_groups = df.drop_duplicates(subset=stat_groups, keep="first").reset_index(
-        drop=True
-    )
-    unique_groups["rolling_window"] = pd.to_datetime(
-        unique_groups["rolling_window"], format="%Y%m"
-    ).dt.date
-
-    groups_string_col = "_".join(map(str, stat_groups))
-    suffixes = ["mean_price"] if condos else ["mean_price", "mean_price_per_sqft"]
-    cols_to_write_means = (
-        stat_groups
-        + ["group_size"]
-        + [f"sv_{suffix}_{groups_string_col}" for suffix in suffixes]
-    )
-
-    rename_dict = {
-        f"sv_{suffix}_{groups_string_col}": f"{suffix}" for suffix in suffixes
-    }
-    df_means = (
-        unique_groups[cols_to_write_means]
-        .rename(columns=rename_dict)
-        .assign(
-            run_id=run_id,
-            group=lambda x: x[stat_groups].astype(str).apply("_".join, axis=1),
-        )
-        .drop(columns=stat_groups)
+    df["group"] = df.apply(
+        lambda row: "_".join([str(row[col]) for col in stat_groups]), axis=1
     )
 
-    # Merge the std and group_mean/size data frames
-    std_cols = ["group", "std"]
-    if not condos:
-        std_cols.append("std_sqft")
-
-    merged_df = df_means.merge(df_std[std_cols], on="group", how="left")
+    if condos:
+        df = df.drop_duplicates(subset=["group"])[
+            ["group", "group_mean", "group_std", "group_size"]
+        ].assign(run_id=run_id)
+    else:
+        df = df.drop_duplicates(subset=["group"])[
+            [
+                "group",
+                "group_mean",
+                "group_std",
+                "group_sqft_std",
+                "group_sqft_mean",
+                "group_size",
+            ]
+        ].assign(run_id=run_id)
```

Comment:
Persisting the standard deviations and group means allowed for a nice refactor to this function.